xonsh/aliases.py

```diff
@@ -254,7 +254,8 @@ def source_cmd(args, stdin=None):
     args[0] = fpath if fpath else args[0]
     if not os.path.isfile(args[0]):
         raise FileNotFoundError(args[0])
-    prevcmd = 'call "{}"\necho off'.format(args[0])
+    prevcmd = 'call "{}"'.format('" "'.join(args))
```

Comment:
I knew that there could be a problem, but didn't know this solution. This may actually be a general problem when xonsh launches subproc on windows. Is this something we have to consider?

 I think don't think we need to think about escaping anything here though, but I do agree that we shouldn't wrap anything in quotes either. The command here is written to a batch file which is then parsed to 'cmd.exe' So the batch file should essentially just contain the same thing that the user wrote in the xonsh shell. We shouldn't try to wrap anything in quotes.